### ISSUE TITLE
Look over latest generated synonyms

### ISSUE THREAD

**Issue by alvaromorales** [infolab-csail/WikipediaBase#132]
I've added diff files to `/data/infolab/misc/WikipediaBase/synonyms/`. The most informative files are `symbols-deleted` and `synonyms-deleted`.

In #121, I made a change that gets rid of many bad synonyms. Here's a summary of my changes:

* Look for "May refer to:" patterns to identify disambiguation pages
* Ignore redirects that point to disambiguation pages
* Ignore symbol chains of length > 3
* Ignore redirects to local anchors
* Require disambiguation links to be only at the beginning of a sentence

Because the generation procedure takes about 40 hours, I ran it against the Dec '15 dump. Previously, synonyms were generated using the Oct '15 dump.

Totals:
* 4625630 symbols (82581 deleted, 59585 new)
* 11661793 synonyms (2892820 deleted, 381574 new)

Some seemingly correct synonyms are also deleted (I say seemingly because though semantically reasonable, they might be generated from an incorrect path along the redirect graph). However, Wikipedia is huge, and I doubt that this will have an impact on actual questions from users. Because we're biased for precision, I think this is the right move. In the future, maybe we can relax some constraints (e.g. lower the priority of synonyms with a chain longer than 3).

**Comment by sfelshin** [comment]
How about checking that all symbols in the symbols file are also in the articles table?  Symbol `10 Janpath (film)` is in `symbols-deleted`.  Maybe the article was deleted between the October and December dumps, but it's worth investigating.

**Comment by sfelshin** [comment]
Nice test case: Deleted symbol `Absolon (surname)` redirects to a fragment anchor `https://en.wikipedia.org/wiki/Absalom_%28name%29#Surname`.

**Comment by alvaromorales** [comment]
I looked into this some more. It turns out that deleted articles are a non-issue.

We've broken up the backend construction into two steps:

1. Process the Wikipedia XML dump and load pages into the articles table.

    We load pages with namespace 0 (articles; includes redirects and disambiguation pages) and namespace 10 (templates).
2. Loop over the articles table to generate symbols and synonyms

Because we generate the symbols from the articles table, all symbols in the symbols file must also be in the articles table. We don't need to add an extra check.

**Comment by sfelshin** [comment]
Okay, @alvaromorales, I'm declaring victory on this.  All START front page tests pass on navassa (except for the usual suspects that fail on the production host, too).  I've looked into about a score of synonym deletions. I've added low-priority issues https://github.com/infolab-csail/WikipediaBase/issues/146 and https://github.com/infolab-csail/WikipediaBase/issues/148 to record a couple of sorts of lost synonyms that would have been nice to keep.
